File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        sort(intervals.begin(), intervals.end());
        int start = intervals[0][0];
        int end = intervals[0][1];

        int row = intervals.size();

        int j = 1;

        vector<vector<int>> ans;

        while (j < row) {

            if (intervals[j][0] <= end) {
                if(intervals[j][1] > end){
                    end = intervals[j][1];
                }
                // end = intervals[j][1];
            }

            else {
                // cout << start << " " << end << endl;
                ans.push_back({start, end});
                start = intervals[j][0];
                end = intervals[j][1];
                // cout << "new start and end => ";
                // cout << start << " " << end << endl;
            }

            j++;
        }

        ans.push_back({start, end});

        return ans;
    }
};

File: 56-merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {

        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));

        List<int[]> merged = new ArrayList<>();

        int start = intervals[0][0];
        int end = intervals[0][1];

        int i = 1;
        int n = intervals.length;

        while (i < n) {

            if (intervals[i][0] > end) {
                int[] temp = new int[2];
                temp[0] = start;
                temp[1] = end;
                merged.add(temp);

                start = intervals[i][0];
                end = intervals[i][1];
            }

            else if (intervals[i][1] > end) {
                end = intervals[i][1];
            }

            i++;

        }

        int[] temp = new int[2];

        temp[0] = start;
        temp[1] = end;
        merged.add(temp);


        return merged.toArray(new int[merged.size()][2]);

    }
}